src/pages/views/PageCreate.tsx

```diff
@@ -51,14 +56,40 @@ export const PageCreate: React.FC<PageCreateProps> = () => {
   return (
     <TypedPageCreate onCompleted={handlePageCreate}>
       {(pageCreate, pageCreateOpts) => {
-        const handleCreate = async (formData: PageData) => {
+        const handleCreate = async (formData: PageSubmitData) => {
+          const uploadFilesResult = await Promise.all(
+            formData.attributesWithNewFileValue.map(fileAttribute =>
+              uploadFile({
+                variables: {
+                  file: fileAttribute.value
+                }
+              })
+            )
+          );
+
+          const attributesWithAddedNewFiles: AttributeValueInput[] = uploadFilesResult.reduce(
```

Comment:
I think since there's no logic that would decide whether to add or not to add item here it could simply be map?